channeldb/codec.go

```diff
@@ -389,7 +400,13 @@ func ReadElement(r io.Reader, element interface{}) error {
 		*e = bytes
 
 	case *lnwire.Message:
-		msg, err := lnwire.ReadMessage(r, 0)
+		var msgLen uint16
+		if err := ReadElement(r, &msgLen); err != nil {
```

Comment:
do we know that `CommitSig` is the only `lnwire.Message` that isn't length-prefixed?

These are all the messages I found passing through this codec in `channeldb` unit tests
```
DECODING CommitSig
DECODING UpdateAddHTLC
DECODING UpdateFailHTLC
DECODING UpdateFulfillHTLC
ENCODING CommitSig
ENCODING UpdateAddHTLC
ENCODING UpdateFailHTLC
ENCODING UpdateFulfillHTLC
```

Reply:
I think one other message is the `ChannelReestablish` that we store within the channel close summary. In the pst I think I had portions of this in this PR, but it's been a bit since I've rebased the PR that adds the actual change in the `lnwire` logic. 

I'm gonna rebase that on top of this, and run the tests there, which in the past have let me catch other issues in this base migration.

Reply:
Oh I think those are the `LogUpdates` in the greater commit diff.

Reply:
`ChannelReestablish` should be handled when we migrate the closed channel summaries using the same construct:
   * read out using the old non-length prefixed version
   * write out using the new length prefixed version